Dockerfile-assemblyscript

```diff
@@ -0,0 +1,20 @@
+FROM node:15.14.0
+LABEL maintainer="Developer Relations Team <support@fastly.com>"
+
+WORKDIR /tmp
+RUN npm install --global npm@latest
+RUN npm install --save-dev assemblyscript
```

Comment:
Why have you done this inside a `/tmp` directory? The `npm install --save-dev assemblyscript` here will likely fail as there isn't an initialised npm package (i.e. a package.json file) to `--save-dev` persist too? A couple of options:
- as the users source code will likely have a valid package.json with `assemblyscript` as a dependency, you could skip it all together
- alternatively you should `npm install --global assemblyscript`, however, you then have the issue of the global version of `asc` being different to that which is defined in the users local package and depending on how their $PATH is configured the global one may be chosen first and lead to confusion.

Reply:
@phamann 

re: 

> Why have you done this inside a /tmp directory? The npm install --save-dev assemblyscript here will likely fail as there isn't an initialised npm package (i.e. a package.json file) to --save-dev persist too? 

This doesn't appear to be the case:

```
root@792eb4caa11c:/tmp# pwd
/tmp
root@792eb4caa11c:/tmp# ls -la
total 5260
drwxrwxrwt 1 root root    4096 Apr 21 14:24 .
drwxr-xr-x 1 root root    4096 Apr 23 14:27 ..
-rw-r--r-- 1 root root 5354637 Apr 21 14:24 fastly.tar.gz
drwxr-xr-x 6 root root    4096 Apr 21 14:24 node_modules
-rw-r--r-- 1 root root    2391 Apr 21 14:24 package-lock.json
-rw-r--r-- 1 root root      64 Apr 21 14:24 package.json
drwxr-xr-x 3 root root    4096 Apr 10 07:39 v8-compile-cache-0
root@792eb4caa11c:/tmp# cat package.json
{
  "devDependencies": {
    "assemblyscript": "^0.18.27"
  }
}
```

I found a `package.json` was generated automatically for me when one wasn't present. 

With that in mind, I realise now that this is redundant because as you also commented...

> as the users source code will likely have a valid package.json with assemblyscript as a dependency, you could skip it all together

This would ultimately mean that the user would need to provide their own AssemblyScript (e.g. via a mounted `node_modules` directory), which is fine if that's the expectation, and additionally it would make the container more flexible as it wouldn't be restricting the version of AssemblyScript a customer wants to utilise.

What about `npm install --global npm@latest`? Is there any concern in having this built into the image? Could this cause conflicts between the user's local code (that they mount into the running container). Maybe we shouldn't do this and just rely on the `FROM node:15.14.0` providing whatever version of `npm` comes with that version of node?